dan-stineback/app/js/game/controllers/game-controller.js

```diff
@@ -0,0 +1,151 @@
+'use strict';
+module.exports = function(app){
+  app.controller('GameController', GameController);
+};
+
+function GameController() {
+  this.userLocation = 'start';
+  this.userHasWeapon = false;
+  this.command = '';
+  this.gamelog = [];
+  this.location = {
+    'start': {
+      commands: ['Enter ? for available commands.'],
+      prompt: 'Sucks to be you. You woke up from a drunken bender in a room with a Vicster.'
+    },
+    'kegroom': {
+      commands: ['take a beer', 'walk through door'],
+      prompt: 'You are in the Keg room. There is a tapped Keg and many Red Solo cups.',
+      promptBeer: 'You have a beer, go throw it in Vicster face'
+    },
+    'vicsterroomwithoutbeer': {
+      commands: ['walk through door', 'ask why he is so sweaty'],
+      prompt: 'You are in a room with Vicster.',
+      promptSweat: 'Fee Fi Foo Fum I Need BEEEEEERRRRR!'
+    },
+    'vicsterroomwithbeer': {
+      commands: ['throw beer', 'walk through door'],
+      prompt: 'You are in a room with Vicster and you have a beer. You now have an aggressive Vicster, The only thing to stop him is beer. Throw it on him!',
+      promptWin: 'You stopped Vicster! You won the game!'
+    }
+  };
+}
+GameController.prototype.startGame = function() {
+  this.gamelog.push({
+    src: 'game',
+    msg: this.location.start.prompt
+  });
+  this.location.start.commands.forEach((item) => {
+    this.gamelog.push({
+      src: 'command',
+      msg: item
+    });
+  });
+  this.userLocation = 'vicsterroomwithoutbeer';
+};
+
+GameController.prototype.processInput = function() {
+
+  this.gamelog.push({
+    src: 'user',
+    msg: this.command
+  });
+
+  switch (this.command) {
+  case '?':
+    this.gamelog.push({
+      src: 'game',
+      msg: this.currentHelpMsg()
+    });
+    break;
+
+  case 'walk through door': {
+    var currentLocation = this.userLocation;
+    if (currentLocation === 'kegroom') {
+      currentLocation = this.userLocation = this.userHasWeapon ? 'vicsterroomwithbeer' : 'vicsterroomwithoutbeer';
+      this.gamelog.push({
+        src: 'game',
+        msg: this.location[currentLocation].prompt
+      });
+    } else {
+      this.userLocation = 'kegroom';
+      this.gamelog.push({
+        src: 'game',
+        msg: this.location.kegroom.prompt
+      });
+    }
+
```

Comment:
example of a refactor to make testing easier:
function walkThroughDoor(userLocation){
  all the code  logic between line 63 AND 75 go here
  return object to push to gamelog
}
Then you test this function instead of ProcessInput function.
This is just an alternative.  I like the behavioral testing that you did though.